addons/stock/models/product.py

```diff
@@ -739,11 +739,17 @@ def write(self, vals):
         if 'type' in vals and vals['type'] != 'product' and sum(self.mapped('nbr_reordering_rules')) != 0:
             raise UserError(_('You still have some active reordering rules on this product. Please archive or delete them first.'))
         if any('type' in vals and vals['type'] != prod_tmpl.type for prod_tmpl in self):
-            existing_move_lines = self.env['stock.move.line'].search([
+            existing_done_move_lines = self.env['stock.move.line'].search([
```

Comment:
@pno-odoo you should add sudo(), to check in multicompany context.

Reply:
You're right. Thank you!